src/Bridge/Doctrine/Common/Filter/BooleanFilterTrait.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Common\Filter;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\DBAL\Types\Type as DBALType;
+
+/**
+ * Trait for filtering the collection by boolean values.
+ *
+ * Filters collection on equality of boolean properties. The value is specified
+ * as one of ( "true" | "false" | "1" | "0" ) in the query.
+ *
+ * For each property passed, if the resource does not have such property or if
+ * the value is not one of ( "true" | "false" | "1" | "0" ) the property is ignored.
+ *
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ * @author Teoh Han Hui <teohhanhui@gmail.com>
+ */
+trait BooleanFilterTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription(string $resourceClass): array
+    {
+        $description = [];
+
+        $properties = $this->properties;
+        if (null === $properties) {
+            $properties = array_fill_keys($this->propertyHelper->getClassMetadata($resourceClass)->getFieldNames(), null);
+        }
+
+        foreach ($properties as $property => $unused) {
+            if (!$this->propertyHelper->isPropertyMapped($property, $resourceClass) || !$this->isBooleanField($property, $resourceClass)) {
+                continue;
+            }
+
+            $description[$property] = [
+                'property' => $property,
+                'type' => 'bool',
+                'required' => false,
+            ];
+        }
+
+        return $description;
+    }
+
+    /**
+     * Determines whether the given property refers to a boolean field.
+     */
+    protected function isBooleanField(string $property, string $resourceClass): bool
```

Comment:
Should be private isn't it (same for all similar methods)?

Reply:
It can, but since the filters are not final, I thought it was for a reason and the methods could be reused. Is there a reason for them to be not final?

Reply:
It was a mistake, it as been forgotten. But it was a while ago, and now it's too late (or we can mark them with `@final` now and `final` in 3.0.). But MongoDB filters, and all new filters, should be `final`.